chain/ethhash/eth_transaction_hash_lookup.go

```diff
@@ -0,0 +1,166 @@
+package ethhash
+
+import (
+	"database/sql"
+	"math"
+
+	"github.com/ipfs/go-cid"
+	_ "github.com/mattn/go-sqlite3"
+	"golang.org/x/xerrors"
+
+	"github.com/filecoin-project/go-state-types/abi"
+
+	"github.com/filecoin-project/lotus/chain/types/ethtypes"
+)
+
+var pragmas = []string{
+	"PRAGMA synchronous = normal",
+	"PRAGMA temp_store = memory",
+	"PRAGMA mmap_size = 30000000000",
+	"PRAGMA page_size = 32768",
+	"PRAGMA auto_vacuum = NONE",
+	"PRAGMA automatic_index = OFF",
+	"PRAGMA journal_mode = WAL",
+	"PRAGMA read_uncommitted = ON",
+}
+
+var ddls = []string{
+	`CREATE TABLE IF NOT EXISTS tx_hash_lookup (
+		hash TEXT PRIMARY KEY UNIQUE,
+		cid TEXT NOT NULL UNIQUE,
+		epoch INT NOT NULL
+	)`,
+
+	`CREATE UNIQUE INDEX IF NOT EXISTS tx_hash_lookup_hash_uindex ON tx_hash_lookup (hash)`,
+
+	`CREATE INDEX IF NOT EXISTS tx_hash_lookup_epoch_index ON tx_hash_lookup (epoch)`,
```

Comment:
I don't know if there are any good uses for storing that index. If we have blocks with transactions, we'll have to calculate the hashes from the messages anyway in the case that they're not in the index. If we do end up storing some mapping of `eth hash` -> `something else`, then it could be useful, but I don't think we have such a mapping right now?